### PR TITLE
fix: issue with kwargs for get_logs using eth-tester

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

somehow the latest web3.py causes intermittent failures with eth-tester get_logs kwargs.
this should take care of it

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Comment by antazoey** [comment]
Conflicting dependency with eth-abi and eip712, introduced here https://github.com/ApeWorX/ape/pull/2002
I have no idea how Ape was able install in CI? 😕

**Comment by antazoey** [comment]
required https://github.com/ApeWorX/eip712/pull/49 i guess

**Comment by antazoey** [comment]
what is going on?!?! cmon

File: setup.py
Comment (fubuloubu) [inline setup.py:1]: Update eip712 pin?

File: src/ape_ethereum/ecosystem.py [hunk 1]
```diff
@@ -935,14 +935,14 @@ def get_abi(_topic: HexStr) -> Optional[LogInputABICollection]:
                     converted_arguments[key] = value
 
             yield ContractLog(
-                block_hash=log["blockHash"],
-                block_number=log["blockNumber"],
+                block_hash=log.get("blockHash", log.get("block_hash")),
+                block_number=log.get("blockNumber", log.get("block_number")),
```
Comment (fubuloubu) [inline src/ape_ethereum/ecosystem.py:939]: To be a little lazier
```suggestion
                block_hash=log.get("blockHash") or log.get("block_hash"),
                block_number=log.get("blockNumber") or log.get("block_number"),
```

**Review by fubuloubu — DISMISSED** [review]

File: src/ape_ethereum/ecosystem.py (hunk 1 above)
Comment (antazoey) [reply src/ape_ethereum/ecosystem.py:939]: i like some of this but maybe not for the ints, as it would do `0 or None` and then be `None` instead of `0`.

Comment (antazoey) [reply src/ape_ethereum/ecosystem.py:939]: or could do  a final `or 0`

Comment (antazoey) [reply src/ape_ethereum/ecosystem.py:939]: and transaction index can be None so, also will keep that in mind

Comment (fubuloubu) [reply src/ape_ethereum/ecosystem.py:939]: what scenario can it be none? when it's pending block?

Comment (antazoey) [reply src/ape_ethereum/ecosystem.py]: oh, I don't know, I was only looking at the type

Comment (antazoey) [reply src/ape_ethereum/ecosystem.py]: yes, pending block, according to the doc str: https://github.com/ApeWorX/ape/blob/main/src/ape/types/__init__.py#L279

**Review by fubuloubu — APPROVED** [review]

**Review by fubuloubu — APPROVED** [review]

**Comment by fubuloubu** [comment]
Looks like test failure is not detecting scripts anymore

**Comment by antazoey** [comment]
somehow there is a weird package called `scripts` in site packages that has now broken our scripting integration

**Comment by antazoey** [comment]
yet, `sys.modules.get("scripts")` is `None`.

**Comment by antazoey** [comment]
This also happens to me on `main` in a fresh env.

**Comment by antazoey** [comment]
Pretty sure i found the culprit: https://github.com/ethereum/web3.py/tree/main/web3/scripts

**Pull request merged**